zerver/views/user_settings.py

```diff
@@ -168,14 +168,18 @@ def json_set_avatar(request, user_profile):
         return json_error("You must upload exactly one avatar.")
 
     user_file = list(request.FILES.values())[0]
-    upload_avatar_image(user_file, user_profile, user_profile.email)
-    do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
-    user_avatar_url = avatar_url(user_profile)
-
-    json_result = dict(
-        avatar_url = user_avatar_url
-    )
-    return json_success(json_result)
+    try:
+        upload_avatar_image(user_file, user_profile, user_profile.email)
+        do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
+        user_avatar_url = avatar_url(user_profile)
+
+        json_result = dict(
+            avatar_url = user_avatar_url
+        )
+        return json_success(json_result)
+    except IOError as e:
+        print(e)
```

Comment:
should replace this print statement with probably a `logging.warning()` with a bit of verbose language about the problem